File: app/src/main/java/com/veganbeauty/admin/data/local/dao/ProductDao.java
package com.veganbeauty.admin.data.local.dao;

import androidx.annotation.Nullable;
import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;
import com.veganbeauty.admin.data.local.entities.ProductEntity;

import java.util.List;

@Dao
public interface ProductDao {

    @Query("SELECT * FROM products")
    LiveData<List<ProductEntity>> getAllLiveData();

    @Query("SELECT * FROM products")
    List<ProductEntity> getAllSync();

    @Query("SELECT * FROM products WHERE id = :id")
    @Nullable
    ProductEntity getByIdSync(String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSync(ProductEntity product);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllSync(List<ProductEntity> products);

    @Update
    void updateSync(ProductEntity product);

    @Delete
    void deleteSync(ProductEntity product);
}

File: app/src/main/java/com/veganbeauty/admin/data/local/dao/ProductDao.kt
package com.veganbeauty.admin.data.local.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.veganbeauty.admin.data.local.entities.ProductEntity

@Dao
interface ProductDao {

    @Query("SELECT * FROM products")
    fun getAllLiveData(): LiveData<List<ProductEntity>>

    @Query("SELECT * FROM products")
    fun getAllSync(): List<ProductEntity>

    @Query("SELECT * FROM products WHERE id = :id")
    fun getByIdSync(id: String): ProductEntity?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertSync(product: ProductEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllSync(products: List<ProductEntity>)

    @Update
    fun updateSync(product: ProductEntity)

    @Delete
    fun deleteSync(product: ProductEntity)
}
